pkg/chunk/composite_store.go

```diff
@@ -174,6 +175,22 @@ func (c compositeStore) GetChunkRefs(ctx context.Context, userID string, from, t
 	return chunkIDs, fetchers, err
 }
 
+func (c compositeStore) GetChunkFetcher(chunk Chunk) *Fetcher {
+	// next, find the schema with the lowest start _after_ through
+	j := sort.Search(len(c.stores), func(j int) bool {
+		return c.stores[j].start > chunk.Through
+	})
+
+	// reduce it by 1 because we want a schema with start <= through
+	j--
+
+	if j >= 0 && len(c.stores) > j {
```

Comment:
```suggestion
	if 0 <= j && j < len(c.stores) {
```

(Same condition, but it's visually easier to see that `j` is bounded by 0 and `len(c.stores)`)